### ISSUE TITLE
LeftWM, fetched as DE instead of WM

### ISSUE THREAD

**Issue by 0323pin** [Macchina-CLI/libmacchina#98]
@grtcdr Hi, hope you're doing fine. Just came across an interesting one :)

Usually I have the DE read-out disabled as I don't use one. Yesterday I started to play arround with `leftwm` and finally got it to build and run from git-HEAD, https://i.postimg.cc/hGksKScV/2021-11-29-231906-1366x768-scrot.png

When `macchina` failed to display the WM read-out, I disabled it and enabled the DE one. Surprise, it actually fetches `leftwm` as DE. Not a big deal, just thought I'd let you know.

**Comment by grtcdr** [comment]
Hey pin!

Sorry for the late answer. Quick question, can you reproduce this with other window managers?

**Comment by 0323pin** [comment]
No worries :)
I can try a few and report back but, I can tell you already now that with `frankenwm` this doesn't happen.

**Comment by grtcdr** [comment]
Well, we can always write a dirty hack to make this work, I'm just afraid the workaround won't be clean. Maybe we can talk to LeftWM's devs about this?

**Comment by 0323pin** [comment]
> Maybe we can talk to LeftWM's devs about this? Rust project to Rust project.

Sure, this is a better option than an ugly hack.
It has been ages since I looked at `neofetch` but, it does fetch it as a WM. Although, this is probably hard coded in there.

Let me test a few WMs and get back to you. I'm now done with packaging `leftwm` and will have some time to try it. I will merge the WM when 0.2.10 comes out, not before that but, I'm using it myself.

**Comment by 0323pin** [comment]
@grtcdr So, I've installed and tested `openbox`, `awesome`, `fluxbox` and `frankenwm`.

All these get fetched as Window Managers. Interestingly, when booting into these, `leftwm` is still shown as Desktop Environment.

**Comment by grtcdr** [comment]
Dang, thanks a lot for testing it on all four!

Seems like LeftWM is setting `XDG_CURRENT_DESKTOP` (and it's not being unset when hopping between other WMs).

I can make an exception for LeftWM, and honestly (I kinda like that it sets this environment variable), if all WMs did this, we wouldn't need `wmctrl`.

**Comment by 0323pin** [comment]
> Seems like LeftWM is setting XDG_CURRENT_DESKTOP

Yes, I did notice this and had already in mind to build `macchina` from source leaving out `wmctrl`, just to get rid of it :)

No rush, at least on NetBSD, I'm the only one using `leftwm`, I haven't merged the package yet and it won't run if you install it with `cargo` because you need to link `libx11` at compile time. The WM will build but, fails to run because it can't find `libx11.so`.

**Comment by 0323pin** [comment]
Oh, just to make you smile ...

```
2021-12-01 23:29 > macchina

  Host         pin@mybox
  Machine      Hewlett-Packard HP EliteBook 820 G1 A3009DD10303
  OS           NetBSD 9.99.92
  DE           LeftWM
  Packages     192 (pkgsrc)
  Terminal     alacritty
  Shell        nu
  Uptime       1h 10m
  CPU          Intel® Core™ i5-4210U CPU @ 1.70GHz (4)
  Memory       1.1 GB/7.5 GB
  Battery      60% & Discharging

/home/pin()
2021-12-01 23:29 >
```

Just need a Rust kernel XD

**Comment by grtcdr** [comment]
That's honestly really cool :heart: 

> Just need a Rust kernel

I'd love to see something built on top of Rust compete with Linux and the BSDs.

**Comment by grtcdr** [comment]
I'm moving this to macchina, since it's not technically a bug, but it's where we can address this issue.
